src/Field/CoordinateReferenceSystemCombo/CoordinateReferenceSystemCombo.tsx

```diff
@@ -55,191 +52,111 @@ export type CRSComboProps = OwnProps & AutoCompleteProps;
-    } = this.state;
-
-    const crsObjects = predefinedCrsDefinitions || crsDefinitions;
-
-    const selected = crsObjects.filter(i => i.code === option.key)[0];
-
-    this.setState({
-      value: selected.value
-    });
-
-    onSelect(selected);
+  const onCrsItemSelect = (_: string, option: DefaultOptionType) => {
+    const selectedProjection = _find(crsObjects, (p: ProjectionDefinition) => p.code === option.code);
+    setSelected(selectedProjection);
+    if (!_isNil(selectedProjection)) {
+      setSearchValue(undefined);
+    }
   };
 
   /**
-   * Tranforms CRS object returned by EPSG.io to antd  Option component
+   * Transform CRS object returned by EPSG.io to antd Option component
    *
-   * @param crsObject Single plain CRS object returned by EPSG.io
+   * @param code The EPSG code of the ProjectionDefinition
+   * @param projDefinition Single plain CRS object returned by EPSG.io
    *
    * @return Option component to render
    */
-  transformCrsObjectsToOptions(crsObject: CrsDefinition) {
-    const value = `${crsObject.value} (EPSG:${crsObject.code})`;
-
+  const transformCrsObjectsToOptions = ([, projDefinition]: [string, ProjectionDefinition]) => {
+    const epsgDescription = getEpsgDescription(projDefinition);
     return (
       <Option
-        key={crsObject.code}
-        value={value}
+        key={projDefinition.code}
+        code={projDefinition.code}
+        value={epsgDescription}
       >
-        {value}
+        {epsgDescription}
       </Option>
     );
-  }
-
-  /**
-   * The render function.
-   */
-  render() {
-    const {
-      className,
-      emptyTextPlaceholderText,
-      onSelect,
-      crsApiUrl,
-      predefinedCrsDefinitions,
-      ...passThroughOpts
-    } = this.props;
-
-    const {
-      crsDefinitions
-    } = this.state;
+  };
 
-    const crsObjects = predefinedCrsDefinitions || crsDefinitions;
+  const onClear = () => {
+    setSelected(undefined);
+    setSearchValue(undefined);
+    setProjectionDefinitions({});
+  };
 
-    const finalClassName = className ? `${className} ${this.className}` : this.className;
+  useEffect(() => {
+    if (!_isNil(epsgIoResults) && !_isEqual(epsgIoResults, projectionDefinitions) && _isNil(selected)) {
+      setProjectionDefinitions(epsgIoResults);
+    }
+  }, [epsgIoResults, projectionDefinitions, selected]);
 
-    return (
-      <AutoComplete
-        className={finalClassName}
-        allowClear={true}
-        onSelect={(v: string, o: any) => this.onCrsItemSelect(v, o)}
-        onChange={(v: string) => this.handleSearch(v)}
-        placeholder={emptyTextPlaceholderText}
-        {...passThroughOpts}
-      >
-        {
-          crsObjects.map(this.transformCrsObjectsToOptions)
-        }
-      </AutoComplete>
-    );
-  }
-}
+  useEffect(() => {
+    if (!_isNil(selected)) {
+      onSelect(selected);
+    }
+  }, [onSelect, selected]);
+
+  const finalClassName = className ? `${className} ${className}` : className;
```

Comment:
this looks not right, isn't `className` duplicated? maybe you meant `defaultClassName`?

Reply:
fixed in 6dc607253a28e1914ffae1fce2b4908db11e133d